File: backend/internal/domain/models/category.go
package models

import "github.com/google/uuid"

type Category struct {
	BaseModel
	Name          string     `gorm:"type:varchar(120);uniqueIndex;not null" json:"name"`
	Description   string     `gorm:"type:text" json:"description"`
	ParentID      *uuid.UUID `gorm:"type:uuid;index" json:"parent_id,omitempty"`
	Parent        *Category  `gorm:"foreignKey:ParentID;constraint:OnUpdate:CASCADE,OnDelete:RESTRICT" json:"parent,omitempty"`
	SubCategories []Category `gorm:"foreignKey:ParentID" json:"sub_categories,omitempty"`
	CreatedBy     uuid.UUID  `gorm:"type:uuid;not null;index" json:"created_by"`
	Creator       User       `gorm:"foreignKey:CreatedBy;constraint:OnUpdate:CASCADE,OnDelete:RESTRICT" json:"creator,omitempty"`
	Documents     []Document `gorm:"foreignKey:CategoryID" json:"-"`
	DocumentCount int64      `gorm:"->"`
}

func (Category) TableName() string {
	return "categories"
}

File: frontend/src/types/category.ts
import type { User } from './auth';

export interface Category {
  id: string;
  name: string;
  description?: string;
  slug?: string;
  parentId?: string | null;
  parent?: Category | null;
  children?: Category[];
  documentCount?: number;
  isActive: boolean;
  createdById?: string;
  createdBy?: User;
  createdAt: string;
  updatedAt: string;
}

export interface CreateCategoryPayload {
  name: string;
  description?: string;
  parentId?: string | null;
  isActive?: boolean;
}

export type UpdateCategoryPayload = Partial<CreateCategoryPayload>;
